ui/vic-ui-h5c/vic/src/vic-app/src/app/services/data-property.service.ts

```diff
@@ -146,33 +146,47 @@ function processVmType(obj: any): any {
         if (obj.powerState === 'poweredOff') {
             return obj;
         }
-    }
-
-    for (let i = 0; i < extConfig.length; i++) {
-        if (extConfig[i].key === VCH_VM_CLIENT_IP_KEY) {
-            const base64_decoded: string = atob(extConfig[i].value);
-            const ipv4: string = base64_decoded.charCodeAt(0) + '.'
-                + base64_decoded.charCodeAt(1) + '.'
-                + base64_decoded.charCodeAt(2) + '.'
-                + base64_decoded.charCodeAt(3);
-            obj.dockerEndpoint = `DOCKER_HOST=tcp://${ipv4}${VCH_VM_ENDPOINT_PORT}`;
-            obj.dockerLog = `https://${ipv4}${VCH_VM_LOG_PORT}`;
-            break;
-        }
-
-        if (extConfig[i].key === CONTAINER_VM_IMAGE_NAME_KEY) {
-            obj.image_name = extConfig[i].value;
-            continue;
-        }
 
-        if (extConfig[i].key === CONTAINER_PRETTY_NAME_KEY) {
-            obj.container_name = extConfig[i].value;
-            continue;
+        let isUsingTls = true;
+
+        for (let { key, value } of extConfig) {
+            if (key === VCH_VM_CLIENT_IP_KEY) {
+                const base64_decoded: string = atob(value);
+                const ipv4: string = base64_decoded.charCodeAt(0) + '.'
+                    + base64_decoded.charCodeAt(1) + '.'
+                    + base64_decoded.charCodeAt(2) + '.'
+                    + base64_decoded.charCodeAt(3);
+                obj.dockerEndpoint = `DOCKER_HOST=tcp://${ipv4}`;
+                obj.dockerLog = `https://${ipv4}${VCH_VM_LOG_PORT}`;
+                continue;
+            }
+
+            if (key === DOCKER_PERSONALITY_ARGS_KEY) {
+                isUsingTls = value.indexOf('2376') > -1;
+                continue;
+            }
         }
 
-        if (extConfig[i].key === CONTAINER_VM_PORTMAPPING_KEY) {
-            obj.portmapping = extConfig[i].value;
-            continue;
+        // since the order in which list items are processed is not much guaranteed,
+        // we set the port for Docker API endpoint at the end of the loop
+        obj.dockerEndpoint += isUsingTls ? ':2376' : ':2375';
+
+    } else {
+        for (let { key, value } of extConfig) {
+            if (key === CONTAINER_VM_IMAGE_NAME_KEY) {
+                obj.image_name = value;
+                continue;
+            }
+
+            if (key === CONTAINER_PRETTY_NAME_KEY) {
+                obj.container_name = value;
+                continue;
+            }
+
+            if (key === CONTAINER_VM_PORTMAPPING_KEY) {
```

Comment:
No, I don't change the container VM ports at all.  Just the container VM ports are managed.